app/models/compute_resources/foreman/model/openstack.rb

```diff
@@ -92,12 +94,37 @@ def boot_from_volume(args = {})
       } ]
     end
 
+    def possible_scheduler_hints
+      SEARCHABLE_ACTIONS.collect{|x| x.to_s.camelize }
+    end
+
+    def get_server_groups(policy)
+      server_groups = client.server_groups.select{ |sg| sg.policies.include?(policy) }
+      errors.add(:scheduler_hint_value, _("No matching servergroups found")) if server_groups.empty?
+      server_groups
+    end
+
+    def format_scheduler_hint_filter(args = {})
+      raise "Hint data is missing" if args[:scheduler_hint_data].nil?
```

Comment:
`raise ::Foreman::Exception.new(N_('Hint data is missing'))` instead, it provides an ERF code and allows us to handle exceptions caused by us.